packages/grz-pydantic-models/src/grz_pydantic_models/submission/metadata/v1.py

```diff
@@ -842,6 +844,90 @@ def ensure_unpaired_long_read_data(self) -> Self:
+        else:
+            if len(bam_files) > 0:
+                raise ValueError("Short-read datasets cannot contain BAM files!")
+            if len(fastq_files) == 0:
+                raise ValueError("Short-read datasets must contain at least one FASTQ file!")
+
+        if self.sequencing_layout == SequencingLayout.paired_end:
+            # check if read order is specified
+            for i in fastq_files:
+                if i.read_order is None:
+                    raise ValueError(
+                        f"Error in lab datum '{self.lab_data_name}': "
+                        f"No read order specified for FASTQ file '{i.file_path}'!"
+                    )
+
+            key = lambda f: (f.flowcell_id, f.lane_id)
+            fastq_files.sort(key=key)
+            for _key, group in groupby(fastq_files, key):
+                flowcell_id = _key[0]
+                lane_id = _key[1]
+                files = list(group)
+
+                # separate R1 and R2 files
+                fastq_r1_files = [f for f in files if f.read_order == ReadOrder.r1]
+                fastq_r2_files = [f for f in files if f.read_order == ReadOrder.r2]
+
+                # check that there are exactly one R1 and on R2 file present
+                if len(fastq_r1_files) > 1:
+                    raise ValueError(
+                        f"Error in lab datum '{self.lab_data_name}': "
+                        f"Paired end sequencing layout but multiple R1 files for flowcell id '{flowcell_id}', lane id '{lane_id}'!"
+                    )
+                elif len(fastq_r1_files) < 1:
+                    raise ValueError(
+                        f"Error in lab datum '{self.lab_data_name}': "
+                        f"Paired end sequencing layout but missing R1 file for flowcell id '{flowcell_id}', lane id '{lane_id}'!"
+                    )
+
+                if len(fastq_r2_files) > 1:
+                    raise ValueError(
+                        f"Error in lab datum '{self.lab_data_name}': "
+                        f"Paired end sequencing layout but multiple R2 files for flowcell id '{flowcell_id}', lane id '{lane_id}'!"
+                    )
+                elif len(fastq_r2_files) < 1:
+                    raise ValueError(
+                        f"Error in lab datum '{self.lab_data_name}': "
+                        f"Paired end sequencing layout but missing R2 file for flowcell id '{flowcell_id}', lane id '{lane_id}'!"
+                    )
+
+        return self
+
+    @model_validator(mode="after")
+    def ensure_unique_run_id(self) -> Self:
+        if not self.sequence_data:
+            return self
+
+        read_files = filter(lambda f: f.file_type in {FileType.fastq, FileType.bam}, self.sequence_data.files)
+        read_files_sorted = sorted(read_files, key=attrgetter("flowcell_id", "lane_id", "read_order"))
+        for (flowcell_id, lane_id, read_order), group in itertools.groupby(
```

Comment:
there's a `from itertools import groupby` import already
```suggestion
        for (flowcell_id, lane_id, read_order), group in groupby(
```